### PR TITLE
feat(core): adds angular 18 support

### LINKED ISSUE #1949: [Feature Request]: Add support for angular 18
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Add support for Angular 18

**Describe alternatives you've considered**
Angular 17 compatible version of the library

**Additional context**
See PR: #1948

### PR DESCRIPTION
**Pull request by pavankjadda**
Closes #1949

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,39 +53,39 @@
...
-    "@angular/router": "^17.0.8",
+    "@angular/animations": "^18.0.1",
+    "@angular/common": "^18.0.1",
+    "@angular/compiler": "^18.0.1",
+    "@angular/core": "^18.0.1",
+    "@angular/forms": "^18.0.1",
+    "@angular/platform-browser": "^18.0.1",
+    "@angular/platform-browser-dynamic": "^18.0.1",
+    "@angular/router": "^18.0.1",
     "@schematics/angular": "^16.0.2",
```
Comment (timdeschryver) [inline package.json:64]: Can you also update this to v18 please

Comment (pavankjadda) [reply package.json]: Done

**Comment by damienbod** [comment]
Thanks @pavankjadda @timdeschryver 

I will try to get this released in the next 2 weeks.

We have a build bug to fix, something with the standalone (not related to this) and then just testing

**Comment by pavankjadda** [comment]
> Thanks @pavankjadda @timdeschryver
> 
> I will try to get this released in the next 2 weeks.
> 
> We have a build bug to fix, something with the standalone (not related to this) and then just testing

Yep. I do see message in v17 and in v18 `callsProvidersFunction()` and `addFunctionalProvidersToStandaloneBootstrap()` are removed
```
Deprecated:
Private utility that will be removed. Use addRootImport or addRootProvider from @schematics/ angular/ utility instead.
```

**Pull request merged**